understanding/20/name-role-value.html

```diff
@@ -14,33 +14,40 @@ <h2>Intent of Name, Role, Value</h2>
       
       
       <p>The intent of this Success Criterion is to ensure that Assistive Technologies (AT)
-         can gather information about, activate (or set) and keep up to date on the status of
+         can gather appropriate information about, activate (or set) and keep up to date on the status of
          user interface controls in the content.
       </p>
       
       <p>When standard controls from accessible technologies are used, this process is straightforward.
          If the user interface elements are used according to specification the conditions
-         of this provision will be met. (See examples  of Success Criterion 4.1.2 below)
+         of this provision will be met. (See examples of Success Criterion 4.1.2 below)
       </p>
       
       <p>If custom controls are created, however, or interface elements are programmed (in
          code or script) to have a different role and/or function than usual, then additional
-         measures need to be taken to ensure that the controls provide important information
+         measures need to be taken to ensure that the controls provide important and appropriate information
          to assistive technologies and allow themselves to be controlled by assistive technologies.
       </p>
+     
+      <p>What roles and states are appropriate to convey to assistive technology will be determined
+         by what the control represents. Specifics about such information are defined by other
+         specifications, such as <a href="https://www.w3.org/TR/wai-aria/">WAI-ARIA</a>, or the
+         appropriate platform standards. Another factor to consider is whether there is sufficient
+         support in assistive technologies to convey the information as specified.
+      </p>
       
       <p>A particularly important state of a user interface control is whether or not it has
          focus. The focus state of a control can be programmatically determined, and notifications
          about change of focus are sent to user agents and assistive technology. Other examples
          of user interface control state are whether or not a checkbox or radio button has
          been selected, or whether or not a collapsible tree or list node is expanded or collapsed.
       </p>
-      
+     
       <div class="note">
          
          <p>Success Criterion 4.1.2 requires a programmatically determinable name for all user
-            interface components. Names may be visible or invisible. Occasionally, the name must
-            be visible, in which case it is identified as a label. Refer to the definition of
+            interface components. Names may be visible or invisible. Occasionally, the name needs
```

Comment:
I'd love to see some about the fact that, normatively, the name doesn't necessarily need to even human-understandable. As for visible/invisible, maybe something a bit more generic about how there is a difference between label and name - that often they may be the same, but not necessarily, and then reference both the glossary definitions AND maybe hint at other SCs like label in name.

Reply:
What do you mean that names don't need to be human-understandable? The name in 4.1.2 is "text by which software can identify a component within Web content to the user"

Reply:
I mean there's no actual requirement in the normative text that that name needs to reflect what the control does, that it's appropriate, understandable. the name could be "xx2" or some nonsense...that is then what the sofware uses to identify the component to the user. there's no qualitative statement about the name in the SC.

Reply:
Sure, but that part is in 1.1.1, right?

Reply:
not necessarily, no - it may fail other criteria of course (as you imply, if it's an image, but then has an accName that is nonsensical like "xx2", it would fail 1.1.1 ... or if it has visible sensible text like "Open the pod bay doors" but has an overriding `aria-label` accName of "xx2" then it would fail 2.5.3 Label in Name). but strictly for 4.1.2, the accName can be anything, doesn't have to be human-understandable (it can pass 4.1.2, but then fail other more specific SCs)

Reply:
Resolving this here for this PR, will open a separate issue/PR for this so as not to hold up this specific one here